pkg/storage/base58wrapper/base58wrapper.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package base58wrapper
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	"github.com/btcsuite/btcutil/base58"
+
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+// package base58wrapper offers provider.Store wrappers for stores with certain constraints on key values such as
+// reserved prefix values (eg key IDs starting with underscore) by wrapping key values in another format supported by
+// the store.
+
+// NewBase58StoreWrapper creates a new Base64ToBase58StoreWrapper of store.
+func NewBase58StoreWrapper(store storage.Store) storage.Store {
+	return &Base64ToBase58StoreWrapper{store: store}
+}
+
+// Base64ToBase58StoreWrapper is a wrapper store that converts key IDs from base64 to base58 encoded values.
+// Keys IDs will be stored in the embedded store base58 encoded while the user of this wrapper store will interact with
+// the wrapper store using key IDs base64 raw (no padding) URL encoded.
+type Base64ToBase58StoreWrapper struct {
+	store storage.Store
+}
+
+func convert(k string) (string, error) {
+	kidBytes, err := base64.RawURLEncoding.DecodeString(k)
```

Comment:
Why not convert base64 string to base58 string right away in order to create new id? I am not sure why you are decoding it first here.

Reply:
Also this is not going to work unless string passed in is base64 encoded

Reply:
I'll update to just base58 encode k directly.

This works because k received is always base64 encoded. The client gets a base64 encoded key IDs always.

Reply:
done